### PR TITLE
DTS Manifest parsing

### PR DESCRIPTION
**Pull request by briehl**
This does the work of parsing the DTS manifest and returning results.
It relies pretty heavily on the jsonschema to do the validation work, so the parser has relatively minimal checking.

Also a minor schema tweak to make sure that parameters are always primitive types.

This is on the big side, but the vast majority is either test data or tests that follow a pretty set pattern.

### CODE REVIEW

File: staging_service/import_specifications/individual_parsers.py [hunk 1]
```diff
@@ -351,15 +357,26 @@ def parse_dts_manifest(path: Path, validator: Draft202012Validator) -> ParseResu
...
             err_path = list(err.absolute_path)
             if err_path:
                 err_str += f" at {err_path}"
             errors.append(Error(ErrorType.PARSE_FAIL, err_str, spcsrc))
+        if not errors:
+            results = defaultdict(list)
+            instructions = manifest[_DTS_INSTRUCTIONS_KEY]
+            for resource_obj in instructions[_DTS_INSTRUCTIONS_OBJECTS_KEY]:
+                datatype = resource_obj[_DTS_INSTRUCTIONS_DATATYPE_KEY]
+                parameters = frozendict(resource_obj[_DTS_INSTRUCTIONS_PARAMETERS_KEY])
```
Comment (MrCreosote) [inline staging_service/import_specifications/individual_parsers.py:374]: So the DTS has to know the app parameter names for each datatype it processes? IOW the DTS is coupled to the apps?

Comment (MrCreosote) [reply staging_service/import_specifications/individual_parsers.py:374]: I also assume the file names / locations in the staging area are in the parameters? They can't be anywhere else...

File: staging_service/import_specifications/individual_parsers.py [hunk 2]
```diff
@@ -351,15 +357,26 @@ def parse_dts_manifest(path: Path, validator: Draft202012Validator) -> ParseResu
...
+                results[datatype].append(parameters)
+            # Re-package results as a dict of {datatype: ParseResult}
+            results = {
+                source: ParseResult(spcsrc, tuple(parsed)) for source, parsed in results.items()
```
Comment (MrCreosote) [inline staging_service/import_specifications/individual_parsers.py]:
```suggestion
                datatype: ParseResult(spcsrc, tuple(paramlist)) for datatype, paramlist in results.items()
```

File: tests/import_specifications/test_individual_parsers.py [hunk 3]
```diff
@@ -806,6 +791,51 @@ def manifest_writer(input_json: dict | list) -> Path:
...
+    f = _get_test_file("manifest_small.json")
+    res = parse_dts_manifest(f, dts_validator)
+    assert res.results
+    assert res.errors is None
+    assert list(res.results.keys()) == ["gff_metagenome"]
+    assert res.results["gff_metagenome"]
+    assert res.results["gff_metagenome"].source.file == f
+    assert len(res.results["gff_metagenome"].result) == 3
+    for parsed in res.results["gff_metagenome"].result:
+        assert parsed == {"param1": "value1", "param2": "value2"}
```
Comment (MrCreosote) [inline tests/import_specifications/test_individual_parsers.py]: Why did you decide to make all the parameters the same?

File: staging_service/import_specifications/individual_parsers.py (hunk 1 above)
Comment (briehl) [reply staging_service/import_specifications/individual_parsers.py:374]: Yes to both. We talked about doing some references from the parameters to file paths, but if we do that'll be in a later version.

Comment (MrCreosote) [reply staging_service/import_specifications/individual_parsers.py:374]: Well, that seems less than ideal, but if you've agreed that that's how it'll work so be it

File: tests/import_specifications/test_individual_parsers.py (hunk 3 above)
Comment (briehl) [reply tests/import_specifications/test_individual_parsers.py]: This was mostly meant as a smoke test / structure test, with more complicated ones elsewhere.
Doesn't matter, though, I updated with more complexity.

File: staging_service/import_specifications/individual_parsers.py (hunk 2 above)
Comment (briehl) [reply staging_service/import_specifications/individual_parsers.py]: changed.

Comment (MrCreosote) [reply staging_service/import_specifications/individual_parsers.py]: you didn't like the paramlist name? That matched up better with the prior var names to me

File: tests/import_specifications/test_individual_parsers.py (hunk 3 above)
Comment (MrCreosote) [reply tests/import_specifications/test_individual_parsers.py]: :+1:

File: staging_service/import_specifications/individual_parsers.py (hunk 2 above)
Comment (briehl) [reply staging_service/import_specifications/individual_parsers.py]: Oops. Meant to change that and got distracted. Changed.

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**